xds/src/main/java/io/grpc/xds/LookasideLb.java

```diff
@@ -437,7 +437,13 @@ public void onEndpointChanged(EndpointUpdate endpointUpdate) {
     @Override
     public void onError(Status error) {
       channelLogger.log(ChannelLogLevel.ERROR, "EDS load balancer received an error: {0}",  error);
-      lookasideLbHelper.updateBalancingState(TRANSIENT_FAILURE, new ErrorPicker(error));
+
+      // If we get an error before getting any valid result, we should put the channel in
+      // TRANSIENT_FAILURE; if they get an error after getting a valid result, we keep using the
+      // previous channel state.
+      if (!firstEndpointUpdateReceived) {
```

Comment:
The state transition is confusing now. If we get an error before LB policy is ready, it first turns channel to TRANSIENT_FAILURE. Then after a while, it goes into fallback mode. Is this intended?

Reply:
It first goes to `TRANSIENT_FAILURE` immediately, and then go to fallback mode. I think this is more correct than silently going to fallback mode. cc @ejona86

Reply:
I sort of expected it to go directly to fallback mode, based on fallback mode in grpclb. At the very least it seems in the code proposed that fallback mode would overwrite the failure here, yet this generally the more important failure.